spec/ddtrace/contrib/faraday/middleware_spec.rb

```diff
@@ -0,0 +1,165 @@
+    end
+  end
+
+  context 'when there is a failing request' do
+    subject!(:response) { client.post('/failure') }
+
+    it do
+      expect(request_span.service).to eq(Datadog::Contrib::Faraday::SERVICE)
+      expect(request_span.name).to eq(Datadog::Contrib::Faraday::NAME)
+      expect(request_span.resource).to eq('POST')
+      expect(request_span.get_tag(Datadog::Ext::HTTP::METHOD)).to eq('POST')
+      expect(request_span.get_tag(Datadog::Ext::HTTP::URL)).to eq('/failure')
+      expect(request_span.get_tag(Datadog::Ext::HTTP::STATUS_CODE)).to eq('500')
+      expect(request_span.get_tag(Datadog::Ext::NET::TARGET_HOST)).to eq('example.com')
+      expect(request_span.get_tag(Datadog::Ext::NET::TARGET_PORT)).to eq('80')
+      expect(request_span.span_type).to eq(Datadog::Ext::HTTP::TYPE)
+      expect(request_span.status).to eq(Datadog::Ext::Errors::STATUS)
+      expect(request_span.get_tag(Datadog::Ext::Errors::TYPE)).to eq('Error 500')
+      expect(request_span.get_tag(Datadog::Ext::Errors::MSG)).to eq('Boom!')
+    end
+  end
+
+  context 'when there is a client error' do
+    subject!(:response) { client.get('/not_found') }
+
+    it { expect(request_span.status).to_not eq(Datadog::Ext::Errors::STATUS) }
+  end
+
+  context 'when there is custom error handling' do
+    subject!(:response) { client.get('not_found') }
+
+    let(:middleware_options) { { error_handler: custom_handler } }
+    let(:custom_handler) { ->(env) { (400...600).cover?(env[:status]) } }
+    it { expect(request_span.status).to eq(Datadog::Ext::Errors::STATUS) }
+  end
+
+  context 'when split by domain' do
+    subject!(:response) { client.get('/success') }
+
+    let(:middleware_options) { { split_by_domain: true } }
+    
+    it do
+      expect(request_span.name).to eq(Datadog::Contrib::Faraday::NAME)
+      expect(request_span.service).to eq('example.com')
+      expect(request_span.resource).to eq('GET')
+    end
+  end
+
+  context 'default request headers' do
+    subject(:response) { client.get('/success') }
+
+    let(:headers) { response.env.request_headers }
+
+    it do
+      expect(headers).to_not include(Datadog::Ext::DistributedTracing::HTTP_HEADER_TRACE_ID)
+      expect(headers).to_not include(Datadog::Ext::DistributedTracing::HTTP_HEADER_PARENT_ID)
+    end
+  end
+
+  context 'when distributed tracing is enabled' do
+    subject(:response) { client.get('/success') }
+
+    let(:middleware_options) { { distributed_tracing: true } }
+    let(:headers) { response.env.request_headers }
+
+    it do
+      expect(headers[Datadog::Ext::DistributedTracing::HTTP_HEADER_TRACE_ID]).to eq(request_span.trace_id.to_s)
+      expect(headers[Datadog::Ext::DistributedTracing::HTTP_HEADER_PARENT_ID]).to eq(request_span.span_id.to_s)
+    end
+
+    context 'but the tracer is disabled' do
+      before(:each) { tracer.enabled = false }
+      it do
+        expect(headers[Datadog::Ext::DistributedTracing::HTTP_HEADER_TRACE_ID]).to be nil
```

Comment:
Maybe you can also write this as
```
expect(headers).to_not include(Datadog::Ext::DistributedTracing::HTTP_HEADER_TRACE_ID)
```
Not really important, but it makes it more consistent with the other expectations you wrote.

Reply:
I like this better.